interfaces/builtin/block_devices.go

```diff
@@ -99,6 +99,9 @@ capability sys_admin,
 # Allow /sys/block/sdX/device/state to be accessible to accept or reject the request from given the path.
 # To take the path offline will cause any subsequent access to fail immediately, vice versa.
 /sys/devices/**/host*/**/state rw,
+
+# Allow to use blkid to export key=value pairs such as UUID to get block device attributes
+/{,usr/}sbin/blkid ixr,
```

Comment:
I don't think that the `r` permission is needed; can you please check if it works without it?

Reply:
It seems to be the default in e.g. `interfaces/apparmor/template.go` so at least there seems to be no harm in adding it.

Reply:
We have some cases of using just `/usr/bin/foo ix,` but most cases we have are `/usr/bin/foo ixr,` like this so I think it's fine, but we should try to be consistent everywhere.

Reading the apparmor manpage though it's unclear that `ix` implies `r`, AFAICT it seems to me that either:
* `ix` implies `r` but this is undocumented
* just using `ix` only allows calling execve on the path, not `read()` on that path, which IMHO is kinda a pointless distinction, I'm not sure why we would only want to allow executing a program and not reading that binary file? Also it feels like executing the program enables reading it anyways too since symbols have to be loaded etc so this could be either not enforced or easily bypassed in practice?

Reply:
`r` has a complicated history. Once upon a time I believe it controlled the dumpable flag on the process, though that doesn't seem to work in a test on 5.4.0-77-generic today. I also think that kernel binfmt elf changes have added and removed the 'read' requirement when executing processes, so it might not be needed in some environments but needed in others.

Sometimes secrets are embedded directly in executables, and the executable readability is used to hide the secret. This is probably not a great design pattern but I understand why folks use it.

I wish I'd kept better notes along the way. Anyway, I tend to use all of `rmix` when I'm writing profiles for myself.

Reply:
@setharnold thanks for the insight; so what I think we should do in this case is use rmix or rix for all binaries like this, but do so in a different PR so that the security team can audit the set of programs that we are allowing with `m` since that flag seems possibly somewhat dangerous for some problematic binaries?

In any case, I don't think that's a reason to hold up this PR which we want to include in an imminent 2.53.2 release, and we have plenty of other programs using `rix`, or more commonly `ixr`, I don't think `blkid` should be subject to additional consideration here regarding `rix` vs `rmix`